src/main/host/descriptor/file.c

```diff
@@ -46,24 +48,33 @@ struct _File {
     /* File is a sub-type of a descriptor. */
     LegacyDescriptor super;
     FileType type;
+    /* A subset of EMULATED_FLAG_MASK */
+    int emulatedFlags;
     /* Info related to our OS-backed file. */
     struct {
         int fd;
-        int flags;
-        mode_t mode;
-        char* abspath;
+        /* A combination of access mode, file creation, and file status flags. */
+        int originalFlags;
```

Comment:
If these represent the flags, mode, and path of the native object, i think `native` would be a more helpful prefix. It's a term we already use elsewhere and emphasizes the contrast with emulatedFlags, etc.

I'm also not sure these prefixes are necessary here, since the field being in the `osfile` or not already imply whether they're the native or emulated versions. I'm ok with or without them, though.

Reply:
These represent the flags, mode, and path of the native object, but only at time of creation. They're not kept up to date as the plugin changes the file flags, mode, and path change, which is why I added the "original" prefix.